backend/internal/service/handler/review.go

```diff
@@ -0,0 +1,104 @@
+package handler
+
+import (
+	"fmt"
+	"platnm/internal/storage"
+	"strconv"
+
+	"github.com/gofiber/fiber/v2"
+)
+
+type ReviewHandler struct {
+	reviewRepository storage.ReviewRepository
+}
+
+func NewReviewHandler(reviewRepository storage.ReviewRepository) *ReviewHandler {
+	return &ReviewHandler{
+		reviewRepository,
+	}
+}
+
+func (h *ReviewHandler) GetReviews(c *fiber.Ctx) error {
+	review, err := h.reviewRepository.GetReviews(c.Context())
+	if err != nil {
+		return err
+	}
+
+	return c.Status(fiber.StatusOK).JSON(review)
+}
+
+func (h *ReviewHandler) GetReviewById(c *fiber.Ctx, mediaType string) error {
+	id := c.Params("id")
+	offsetstr := c.Query("offset", "0")
+	limitstr := c.Query("limit", "10")
+	//limitParam := r.URL.Query().Get("limit")
+	//offsetParam := r.URL.Query().Get("offset")
+	//review, err := h.reviewRepository.GetReviewsByID(id, mediaType, c.Context())
+
+	var offset int
+	if offsetstr == "" {
+		offset = 0 // Default to 0 if no offset is provided
+	} else {
+		var err error
+		offset, err = strconv.Atoi(offsetstr)
+		if err != nil || offset < 0 {
+			offset = 0 // Ensure offset is non-negative
+		}
+	}
+
+	// Check if limitstr is empty, then default to 10
+	var limit int
+	if limitstr == "" {
+		limit = 10 // Default to 10 if no limit is provided
+	} else {
+		var err error
+		limit, err = strconv.Atoi(limitstr)
+		if err != nil || limit <= 0 {
+			limit = 10 // Ensure limit is positive
+		}
+	}
+
+	print("running?")
+
+	// Fetch the review based on ID and media type
+	review, err := h.reviewRepository.GetReviewsByID(c.Context(), id, mediaType)
+	if err != nil {
+		// If error, log it and return 500
+		fmt.Println(err.Error(), "from transactions err ")
+		return c.Status(fiber.StatusInternalServerError).JSON(fiber.Map{
+			"error": "Unable to retrieve reviews",
+		})
+	}
+
+	// If no review is found, return a 404 status
+	if len(review) == 0 {
```

Comment:
for no reviews found, prob should just return empty array since it is okay for a track/album to have no reviews. i think you would wanna return 404 not found if album id or track id does not exist